ksqldb-engine/src/main/java/io/confluent/ksql/util/TransientQueryMetadata.java

```diff
@@ -75,18 +74,28 @@ public TransientQueryMetadata(
         kafkaStreamsBuilder,
         streamsProperties,
         overriddenProperties,
-        closeCallback,
         closeTimeout,
         new QueryId(queryApplicationId),
         QueryErrorClassifier.DEFAULT_CLASSIFIER,
         maxQueryErrorsQueueSize,
         retryBackoffInitialMs,
-        retryBackoffMaxMs
+        retryBackoffMaxMs,
+        listener
     );
-    this.initialize();
```

Comment:
I just had the registry call `initialize` for all queries - I think it's easier to reason about this code if we keep the behavior the methods common to PersistentQueryMetadata and TransientQueryMetadata (constructor, initialize, start, close) the same, and make any differences explicit in the API (e.g. by moving stop to PersistentQueryMetadata)

Reply:
SGTM.